template/module/static/src/js/module_name.js

```diff
@@ -1,7 +1,20 @@
 /* Copyright <YEAR(S)> <AUTHOR(S)>
  * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
 
-(function ($) {
+odoo.define('module_name.object_name', function(require){
     "use strict";
-    // Script that will be loaded when document is ready
-})(jQuery);
+
+    var core = require('web.core');
+
+    core.bus.on('web_client_ready', null, function () {
+        // Script that will be loaded when document is ready
+    });
```

Comment:
I see no reason to not include as many examples as possible. @pedrobaeza you're thinking a separate file would be good here instead of just putting both here and commenting the difference?